jupyter-images/gpu/Dockerfile

```diff
@@ -26,12 +26,19 @@ RUN apt-get update --yes && \
     pip install --no-cache-dir jupyterhub==3.0.0 && \
     pip install --no-cache-dir --upgrade jupyterlab notebook jupyter_server && \
     python3 -m pip install cartopy catboost metpy minisom netCDF4 pillow \
-    pyvista seaborn shapely torch torchaudio torchvision verde xarray \
+    seaborn shapely torch torchaudio torchvision verde xarray \
     --extra-index-url https://download.pytorch.org/whl/cu112 && \
     apt-get clean && rm -rf /var/lib/apt/lists/* && \
     echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && \
     locale-gen
 
+# Handle PyVista on its own
+# Be sure not to pull `vtk`, but `vtk-osmesa` from Kitware's package index
+RUN python3 -m pip install --no-cache-dir --upgrade \
+    pyvista pyvista-xarray jupyter-server-proxy trame && \
+    python3 -m pip uninstall vtk -y && \
+    python3 -m pip install --no-cache-dir --upgrade --extra-index-url https://wheels.vtk.org vtk-osmesa
```

Comment:
Thought I'd elaborate on this.

PyVista forcefully pulls `vtk` as a dependency from PyPI. This is best for *most* users. However, for JupyterHub deployments, we typically want to use an OSMesa or EGL variant of VTK. These variants on not on PyPI but on a custom package index managed by Kitware (https://wheels.vtk.org). I have some more details in this post: https://discourse.vtk.org/t/status-update-vtk-python-wheels/11212

So here, we install PyVista which pulls VTK. Then we have to full uninstall that version of VTK to then install the proper `vtk-osmesa` variant. `vtk` and `vtk-osmesa` are totally separate distributions and cannot be both be installed at the same time